### ISSUE TITLE
bilinear_sampler / coords_grid swap the H and D axes

### ISSUE THREAD

**Issue by zachtong** [hereon-mbs/VolRAFT#1]
The `# Need to check whether it is necessary` note in `coords_grid` turns out to
matter. The released sampler reads the correlation volume with the H and D axes
swapped, and on non-cubic volumes the normalisation is off as well.

Checked on `1dd7005` (current `main`).

### What happens

`coords_grid` (`models/VolRAFT/utils/utils.py:81`) stacks the channels as
`(h, w, d)`. `bilinear_sampler` (`:57`) splits them into `xgrid, ygrid, zgrid`
and normalises by `(W-1)`, `(H-1)`, `(D-1)` in that order, then passes them to
`F.grid_sample`.

For a 5-D input, `grid_sample` reads the last grid dimension as `(x, y, z)`
indexing the **last, second-last and third-last** tensor dims, i.e. `D, W, H`.
So channel 0 (`h`) ends up driving the D axis and channel 2 (`d`) the H axis.

### Reproducer

Identity sampling: build a non-cubic volume with a single impulse, query it at
its own coordinates with the upstream sampler. It should come back unchanged.

```python
import sys, torch
sys.path.insert(0, "models/VolRAFT")
from utils.utils import bilinear_sampler, coords_grid

H, W, D = 6, 8, 10
h0, w0, d0 = 2, 3, 5
img = torch.zeros(1, 1, H, W, D)
img[0, 0, h0, w0, d0] = 1.0

coords = coords_grid(1, H, W, D, device="cpu").permute(0, 2, 3, 4, 1)  # as volcorr.py:32
out = bilinear_sampler(img, coords)

print("value at the query coordinate:", out[0, 0, h0, w0, d0].item())
k = torch.argmax(out.abs())
print("impulse lands at:", torch.unravel_index(k, out.shape)[2:])
```

Output:

```
value at the query coordinate: 0.0
impulse lands at: (4, 2, 4)
```

### Why it isn't obvious

On a cubic volume the three denominators are equal, so the only error left is the
H↔D transpose: a fixed relabelling of the correlation field that a network can
partly absorb while training. Since training and inference both run on cubic
volumes, nothing looks broken until the input is not cubic.

Worth noting: un-commenting `coords = torch.stack(coords[::-1], dim=0).float()`
(`:90`) is not sufficient. It fixes the pairing, so the impulse comes back
correctly on a cubic volume, but on `(6,8,10)` it still lands at `(4,2,4)`
because each coordinate is still normalised by the wrong extent.

### Suggested fix

Pair each coordinate with the `grid_sample` slot that indexes its own axis and
normalise it by its own extent, leaving `coords_grid` as it is:

```python
hgrid, wgrid, dgrid = coords.split([1, 1, 1], dim=-1)
xgrid = 2 * dgrid / (D - 1) - 1   # x indexes the last dim      (D)
ygrid = 2 * wgrid / (W - 1) - 1   # y indexes the second-last   (W)
zgrid = 2 * hgrid / (H - 1) - 1   # z indexes the third-last    (H)
grid = torch.cat([xgrid, ygrid, zgrid], dim=-1)
```

With this the impulse test returns 1.0 at the query coordinate for cubic and
non-cubic volumes alike. The `mask` branch needs no change, since it reads the
normalised grids.

### Context

We ran into the same mistake in our own 3-D RAFT port, which is what made us
check here. Retraining VolRAFT on our synthetic data with the corrected sampler
improved accuracy at the native input size and much more so at input sizes not
seen during training. The impulse test and the numbers are in
https://arxiv.org/abs/2609.01876 and https://github.com/zachtong/RAFT-DVC.

Glad to open a PR if that's easier.
